<launch>
  <!--This launch file is used on the robot side, and only activates the data_throttle node to receive raw data to pass over through the bandwidth-->
  <!--Launch file arguments.
        One thing to be considered is to turn on the laser apart with rosrun and then launching this launch file so as to avoid "unknown error" with hokuyo.
  -->
  <arg name="sensor_on" default="false"/>
  <arg name="rviz" default="false"/>


  <!--Launch sensors for unitary test-->
  <group if="$(arg sensor_on)">
        <include file="$(find openni2_launch)/launch/openni2.launch">
            <arg name="depth_registration" default="true"/>
        </include>
  </group>


  <!-- Throttling messages -->
  <group ns="camera">
    <node pkg="nodelet" type="nodelet" name="data_throttle" args="load rtabmap_ros/data_throttle camera_nodelet_manager" output="screen">
      <param name="max_rate" type="double" value="5.0"/>

       <!--
      <remap from="rgb/image_in"       to="/data_throttled_image"/>
      <remap from="depth/image_in"     to="/data_throttled_image_relay"/>
      <remap from="rgb/camera_info_in" to="/data_throttled_camera_info_relay"/>
      -->
      <remap from="rgb/image_in"       to="/rgb/image_rect_color"/>
      <remap from="depth/image_in"     to="/depth_registered/image_raw"/>
      <remap from="rgb/camera_info_in" to="/rgb/camera_info"/>

      <remap from="rgb/image_out"       to="/data_throttled_image"/>
      <remap from="depth/image_out"     to="/data_throttled_image_depth"/>
      <remap from="rgb/camera_info_out" to="/data_throttled_camera_info"/>
    </node>

    <!-- My own topic_tools launch for Spencer. Output:: /camera/data_throttled_depth_image (don't get confused with /camera/data_throttled_image_depth)-->
     <!-- Sampling rate at 1 Hz -->
     <node pkg="topic_tools" 
                type="throttle" 
                name="my_camera_throttle" 
                arg="messages /camera/depth/image 1 /camera/data_throttled_depth_image" 
                output="screen">
     </node>
     
  </group>

  <!--Launch Rviz-->
  <group if="$(arg rviz)">
    <node pkg="rviz" type="rviz" name="rviz"/>
  </group>
</launch>
